test/mpi/distributed/vector.cpp

```diff
@@ -0,0 +1,545 @@
+        std::shared_ptr<const gko::Executor> exec)
+    {
+        return std::unique_ptr<HostToDeviceLogger>(
+            new HostToDeviceLogger(std::move(exec)));
+    }
+
+protected:
+    explicit HostToDeviceLogger(std::shared_ptr<const gko::Executor> exec)
+        : gko::log::Logger(exec, gko::log::Logger::copy_started_mask)
+    {}
+
+private:
+    mutable int transfer_count_ = 0;
+};
+
+
+class VectorReductions : public ::testing::Test {
+public:
+    using value_type = float;
+    using local_index_type = gko::int32;
+    using global_index_type = gko::int64;
+    using part_type =
+        gko::distributed::Partition<local_index_type, global_index_type>;
+    using md_type = gko::matrix_data<value_type, global_index_type>;
+    using dist_vec_type = gko::distributed::Vector<value_type>;
+    using dense_type = gko::matrix::Dense<value_type>;
+
+    VectorReductions()
+        : ref(gko::ReferenceExecutor::create()),
+          exec(),
+          comm(MPI_COMM_WORLD),
+          size{53, 11},
+          x(dist_vec_type::create(ref, comm)),
+          dx(dist_vec_type::create(exec, comm)),
+          y(dist_vec_type::create(ref, comm)),
+          dy(dist_vec_type::create(exec, comm)),
+          logger(gko::share(HostToDeviceLogger::create(exec))),
+          engine(42)
+    {
+        init_executor(ref, exec, comm);
+        exec->add_logger(logger);
+
+        auto num_parts =
+            static_cast<gko::distributed::comm_index_type>(comm.size());
+        auto mapping =
+            gko::test::generate_random_array<gko::distributed::comm_index_type>(
+                size[0],
+                std::uniform_int_distribution<
+                    gko::distributed::comm_index_type>(0, num_parts - 1),
+                engine, ref);
+        auto part = part_type::build_from_mapping(ref, mapping, num_parts);
+
+        auto md_x = gko::test::generate_random_matrix_data<value_type,
+                                                           global_index_type>(
+            size[0], size[1],
+            std::uniform_int_distribution<gko::size_type>(size[1], size[1]),
+            std::normal_distribution<gko::remove_complex<value_type>>(),
+            engine);
+        x->read_distributed(md_x, part.get());
+        dx = gko::clone(exec, x);
+
+        auto md_y = gko::test::generate_random_matrix_data<value_type,
+                                                           global_index_type>(
+            size[0], size[1],
+            std::uniform_int_distribution<gko::size_type>(size[1], size[1]),
+            std::normal_distribution<gko::remove_complex<value_type>>(),
+            engine);
+        y->read_distributed(md_y, part.get());
+        dy = gko::clone(exec, y);
+    }
+
+    void SetUp() override { ASSERT_EQ(comm.size(), 3); }
+
+    void init_result()
+    {
+        res = dense_type::create(ref, gko::dim<2>{1, size[1]});
+        dres = dense_type::create(exec, gko::dim<2>{1, size[1]});
+        res->fill(0.);
```

Comment:
nit: we usually write them out
```suggestion
        res->fill(0.0);
```